opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ContactSummaryInteractor.java

```diff
@@ -45,63 +45,57 @@ public void fetchWomanDetails(String baseEntityId, BaseContactContract.Interacto
-
-                    Map<String, String> clientDetails =
-                            AncLibrary.getInstance().getDetailsRepository().getAllDetailsForClient(entityId);
-                    JSONObject rawContactSchedule;
-                    if (clientDetails.containsKey(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE)) {
-                        rawContactSchedule = new JSONObject(
-                                AncLibrary.getInstance().getDetailsRepository().getAllDetailsForClient(entityId)
-                                        .get(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE));
-                    } else {
-                        rawContactSchedule = new JSONObject();
-                    }
+        Runnable runnable = () -> {
+            try {
+                Map<String, String> details = PatientRepositoryHelper.getWomanProfileDetails(entityId);
+
+                Map<String, String> clientDetails =
+                        AncLibrary.getInstance().getDetailsRepository().getAllDetailsForClient(entityId);
+                JSONObject rawContactSchedule;
+                if (clientDetails.containsKey(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE)) {
+                    rawContactSchedule = new JSONObject(
+                            AncLibrary.getInstance().getDetailsRepository().getAllDetailsForClient(entityId)
+                                    .get(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE));
+                } else {
+                    rawContactSchedule = new JSONObject();
+                }
 
-                    List<String> contactSchedule = new ArrayList<>();
-                    if (StringUtils.isEmpty(referralContactNo)) {
-                        if (rawContactSchedule.has(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE)) {
-                            contactSchedule =
-                                    Utils.getListFromString(
-                                            rawContactSchedule.getString(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE));
-                        }
-                    } else {
-                        int previousContact = getPreviousContactNo(referralContactNo);
-                        if (previousContact > 0) {
-                            Facts facts =
-                                    AncLibrary.getInstance().getPreviousContactRepositoryHelper()
-                                            .getImmediatePreviousSchedule(entityId, String.valueOf(previousContact));
-                            if (facts != null && facts.asMap().containsKey(ConstantsUtils.CONTACT_SCHEDULE)) {
-                                String schedule = (String) facts.asMap().get(ConstantsUtils.CONTACT_SCHEDULE);
-                                contactSchedule = Utils.getListFromString(schedule);
-                            }
-                        }
+                List<String> contactSchedule = new ArrayList<>();
+                if (StringUtils.isEmpty(referralContactNo)) {
```

Comment:
please use `StringUtils.isBlank(referralContactNo)` instead of `StringUtils.isEmpty(referralContactNo)`.... the latter does not check for whitespace characters